Java/README.md

```diff
@@ -1,12 +1,50 @@
 # Wikia's Java Coding Guidelines
 
+All code should be developed for, and compiled with Java 8.
+
+Our standard package prefix is com.wikia
+
+## Syntax and Formatting
+
 Wikia follows [Google guidelines](https://google-styleguide.googlecode.com/svn/trunk/javaguide.html)
 
 With following addition:
  - 100 column limit should be used by all projects, with exceptions noted in Google guidelines.
 
+## Coding Conventions
+
+In addition to the conventions described in the Google standards, 
+
+### Avoid checked exceptions
+
+Checked exceptions are an interesting experiment that didn't really work out.
+
+The only exception (sorry) to this rule is if you are wrapping or extending some standard Java functionality that throws one of the well-known checked exceptions (IOException, JDBCException, etc) and you want to propagate that exception to callers that are already probably being forced to handle those exceptions by the standard library.
+
+### Do not return null or accept null as an argument
+
+If you want to represent a value that may or may not be present, make it explicit by using java.util.Optional.
 
-## InteliJ configuration:
+Avoid using Optional.get() and Optional.isPresent(). Prefer map() and flatMap() for mutating optionals, and orElseGet() and orElseThrow() for flattening them.
+
+## API Conventions
+
+In addition to other conventions, the following rules apply to APIs that are to be exposed to third party developers (i.e. through being open sourced).
+
+### No dependencies on third-party classes
+
+Public APIs should only expose classes or methods controlled by Wikia, or that are part of the Java standard library. Exposing third-party classes or methods in APIs will inevitably make it impossible to upgrade the third-party library without breaking the API.
+
+## Standard Libraries/Tools
+
+* *Logging:* [SLF4J](http://www.slf4j.org) is the standard logging API. Unless another is required by a services's framework/server, the slf4j-log4j bridge should be used as the back-end.
```

Comment:
My vote would be for Logback as sl4j backend.
+ it has some nice logstash integration: https://github.com/logstash/logstash-logback-encoder